File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.java
class Solution{
    boolean solve(int n){
        int temp=n;
        int sum=0;
        int prod=1;

        while(n>0){
            int rem=n%10;
            n/=10;
            sum += rem;
            prod *= rem;
        }

        int total=sum+prod;
        if(temp%total == 0)
            return true;
        
        return false;
    }

    public boolean checkDivisibility(int n){
        
        return solve(n);


    }
}

File: Easy/3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def solve(self, n):
        temp=n
        s=0
        prod=1

        while n>0:
            rem=n%10
            n//=10
            s += rem
            prod *= rem

        total=s+prod
        if temp%total == 0:
            return True
        
        return False
    
    def checkDivisibility(self, n: int) -> bool:
        return self.solve(n)

